client/my-sites/hosting/main.js

```diff
@@ -184,207 +176,173 @@ const SidebarCards = ( { isBasicHostingDisabled } ) => {
-			event.preventDefault();
-			event.stopImmediatePropagation();
-			this.setState( { clickOutside: true } );
-		}
-	}
-
-	componentDidMount() {
-		const { COMPLETE } = transferStates;
-		// Check if a reverted site still has the COMPLETE status
-		if ( this.props.transferState === COMPLETE ) {
-			// Try to refresh the transfer state
-			this.props.fetchAutomatedTransferStatus( this.props.siteId );
-		}
+const Hosting = ( props ) => {
+	const {
+		teams,
+		clickActivate,
+		isECommerceTrial,
+		isBusinessTrial,
+		isWpcomStagingSite,
+		siteId,
+		siteSlug,
+		translate,
+		isLoadingSftpData,
+		hasAtomicFeature,
+		hasSftpFeature,
+		hasStagingSitesFeature,
+		isJetpack,
+	} = props;
+
+	const { isTransferring, transferStatus } = useAtomicTransferQuery( siteSlug, {
+		refetchInterval: 5000,
+	} );
+
+	const hasFetchedTransferStatus = transferStatus !== undefined;
+	const hasFetchedTransferStatusAtLeastOnce = useRef( false );
+	if ( hasFetchedTransferStatus ) {
+		hasFetchedTransferStatusAtLeastOnce.current = true;
 	}
 
-	render() {
-		const {
-			teams,
-			clickActivate,
-			isAdvancedHostingDisabled,
-			isBasicHostingDisabled,
-			isECommerceTrial,
-			isMigrationTrial,
-			isHostingTrial,
-			isSiteAtomic,
-			isTransferring,
-			isWpcomStagingSite,
-			requestSiteById,
-			siteId,
-			siteSlug,
-			translate,
-			transferState,
-			isLoadingSftpData,
-			hasAtomicFeature,
-			hasStagingSitesFeature,
-			isJetpack,
-		} = this.props;
-
-		const getUpgradeBanner = () => {
-			// The eCommerce Trial requires a different upsell path.
-			const targetPlan = ! isECommerceTrial
-				? undefined
-				: {
-						callToAction: translate( 'Upgrade your plan' ),
-						feature: FEATURE_SFTP_DATABASE,
-						href: `/plans/${ siteSlug }?feature=${ encodeURIComponent( FEATURE_SFTP_DATABASE ) }`,
-						title: translate( 'Upgrade your plan to access all hosting features' ),
-				  };
-
-			return <HostingUpsellNudge siteId={ siteId } targetPlan={ targetPlan } />;
-		};
-
-		const { COMPLETE, FAILURE } = transferStates;
-
-		const isTransferInProgress = () => isTransferring && COMPLETE !== transferState;
-
-		const getAtomicActivationNotice = () => {
-			// Transfer in progress
-			if ( isTransferInProgress() || ( isBasicHostingDisabled && COMPLETE === transferState ) ) {
-				if ( COMPLETE === transferState ) {
-					requestSiteById( siteId );
-				}
-
-				let activationText = translate( 'Please wait while we activate the hosting features.' );
-				if ( this.state.clickOutside ) {
-					activationText = translate( "Don't leave quite yet! Just a bit longer." );
-				}
-
-				return (
-					<>
-						<Notice
-							className="hosting__activating-notice"
-							status="is-info"
-							showDismiss={ false }
-							text={ activationText }
-							icon="sync"
-						/>
-					</>
-				);
-			}
+	const isSiteAtomic =
+		transferStatus === transferStates.COMPLETE || transferStatus === transferStates.COMPLETED;
```

Comment:
Does that necessarily mean the site is Atomic? There are several ways to check if the site is Atomic, not necessarily relying on the transfer status:
- `is_automated_transfer` site option
- `is_wpcom_atomic` site option (seems to be the most reliable and tied to WPCOM)

Reply:
The `complete[d]` state does mean the site is Atomic. If you revert the site, then that also uses a transfer, which changes the status to `reverted`.  If the site is Simple and never went Atomic there will be no transfer. If an error happens, the status will be `error`. There are some other states in `transferStates` to cover other scenarios. 

There are other flags that require a different query, whereas we get the final status + interim status updates via this one query.

Reply:
Ok. I'm still skeptical about data derivation (Atomic meaning the site has transfer state), but it worked in testing, so why not try it as the feature was broken for a while.

I'm considering a case where the transfer was made a while ago and the transfer record has been cleaned up. In this case there won't be a transfer. So WDYT of adding `|| site.options.is_wpcom_atomic` as a final check at the end, just to be sure?